nibabel/tests/test_volumeutils.py

```diff
@@ -733,9 +739,10 @@ def test_apply_scaling():
     f32_arr = np.zeros((1,), dtype=f32)
     i16_arr = np.zeros((1,), dtype=np.int16)
     # Check float upcast (not the normal numpy scalar rule)
-    # This is the normal rule - no upcast from scalar
-    assert (f32_arr * f64(1)).dtype == np.float32
-    assert (f32_arr + f64(1)).dtype == np.float32
+    # This is the normal rule - no upcast from Python scalar
+    # (on NumPy 2.0 it *will* upcast from a np.float64 scalar!)
+    assert (f32_arr * 1.).dtype == np.float32
+    assert (f32_arr + 1.).dtype == np.float32
```

Comment:
The goal here seems to be contrasting to numpy casting rules, not numpy + Python scalar casting rules. I would just branch on numpy version about the expected dtype for using `f64(...)`.